src/telemetry/telemetry.c

```diff
@@ -249,16 +249,21 @@ build_version_request(const char *host, const char *path)
 Connection *
 telemetry_connect(const char *host, const char *service)
 {
-	Connection *conn;
+	Connection *conn = NULL;
 	int			ret;
 
-	conn = connection_create(CONNECTION_SSL);
+	if (strcmp("http", service) == 0)
+		conn = connection_create(CONNECTION_PLAIN);
+	else if (strcmp("https", service) == 0)
+		conn = connection_create(CONNECTION_SSL);
+	else
+		ereport(WARNING,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("scheme \"%s\" not supported for telemetry",
+						service)));
 
 	if (conn == NULL)
-	{
-		elog(WARNING, "could not create telemetry connection");
```

Comment:
I left this warning message (and the one below inside `telemetry_main`), because even though there will be error messages inside the `conn` library, upper-level error messages can be helpful for debugging.
Having both messages is redundant, but maybe leaving one of them in helps future debugging?